test/advection_test/test_host.F90

```diff
@@ -33,15 +33,19 @@ module test_prog
 
 CONTAINS
 
-   subroutine check_errflg(subname, errflg, errmsg)
+   subroutine check_errflg(subname, errflg, errmsg, errflg_final)
       ! If errflg is not zero, print an error message
-      character(len=*), intent(in) :: subname
-      integer,          intent(in) :: errflg
-      character(len=*), intent(in) :: errmsg
+      character(len=*), intent(in)    :: subname
+      integer,          intent(in)    :: errflg
+      character(len=*), intent(in)    :: errmsg
+
+      integer,          intent(inout) :: errflg_final
```

Comment:
Should this be just a `intent(out)` declaration instead of `intent(inout)` as we only write to it here?

Reply:
Good catch!  Should be fixed now.